flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceSplitReader.java

```diff
@@ -113,7 +139,13 @@ public void handleSplitsChanges(SplitsChange<IcebergSourceSplit> splitsChange) {
   }
 
   @Override
-  public void wakeUp() {}
+  public void wakeUp() {
+    LOG.info("WakeUp called");
+    pausedSplits.clear();
+    synchronized (signal) {
+      signal.notify();
```

Comment:
probably doesn't matter. but is it safer to use `notifyAll` here?

Reply:
Done, but not really needed